eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -610,6 +610,8 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, remo
         case Success(_) =>
           log.info(s"channelId=${commitments.channelId} was confirmed at blockHeight=$blockHeight txIndex=$txIndex")
           blockchain ! WatchFundingLost(self, commitments.commitInput.outPoint.txid, nodeParams.minDepthBlocks)
+          if (!d.commitments.localParams.isFunder) context.system.eventStream.publish(TransactionPublished(commitments.channelId, remoteNodeId, fundingTx, 0 sat, "funding"))
+          context.system.eventStream.publish(TransactionConfirmed(commitments.channelId, remoteNodeId, fundingTx))
```

Comment:
`0 sat`: I see what you did there!

So we are also tracking txes that we didn't publish ourselves. It's a change of semantics, but probably okay, and I see how it can be useful.

Reply:
Exactly, the idea is to track transactions that are important for the channel logic, whether we published them or not.
So we're storing their 2nd-stage txs as well once they confirm, which can make it easy to see which outputs they got and which we got.